### DISCUSSION TITLE
Checking if an Object Belongs to an Encrypted 3D Component

### DISCUSSION THREAD

**Discussion by chamanth-vct** [ansys/pyaedt#5398] [Q&A]
Hi,

I have a question regarding handling encrypted 3D components within a model. Currently, when a model includes encrypted 3D components, the names of the objects within these components are available in the list of object names, as demonstrated by the code snippet below:

```py
desktop = ansys.aedt.core.Desktop() 
prj= desktop.load_project(file_path)
model = prj.modeler
object_names = model.object_names
```

Is there a built-in method or function to identify if an object is part of an encrypted 3D component? Any guidance or suggestions would be greatly appreciated.

Thanks

**Comment by fwlfHS** [comment]
you could get the names of the 3d components: [user_defined_component_names](https://aedt.docs.pyansys.com/version/stable/API/_autosummary/ansys.aedt.core.modeler.modeler_3d.Modeler3D.user_defined_component_names.html)
and get all belonging objects: [get_3d_component_object_list](https://aedt.docs.pyansys.com/version/stable/API/_autosummary/ansys.aedt.core.modeler.modeler_3d.Modeler3D.get_3d_component_object_list.html)
now the open topic is still if a component is encrypted or not. Not sure if you can get this information through the API

**Reply by Samuelopez-ansys** [reply]
I agree with your answer. About encrypted components, you are right, there is no way to retrieve this information

**Comment by chamanth-vct** [comment]
Hi @fwlfHS and @Samuelopez-ansys,

Thanks for your response.

We cannot create mesh plots or field plots for objects within encrypted components, and attempting to do so results in errors. To manage this, it's important to identify and filter these encrypted components before generating such plots. This is why I posed the original question. Is there any workaround for this?
